CONTRIBUTING.md

```diff
@@ -116,7 +116,7 @@ cargo 1.69.0 (6e9a83356 2023-04-12)
 
 Currently, iceberg-rust uses Docker to set up environment for integration tests. Native Docker has some limitations, please check (https://github.com/apache/iceberg-rust/pull/748). Please use Orbstack or Podman.
 
-For MacOS users, you can install [OrbStack](https://orbstack.dev/) as a docker alternative.
+For MacOS users, you can install [OrbStack as a docker alternative](docs/contributing/orbstack.md).
```

Comment:
Adding guide about how to use a tool seems strange to me. It should be enough to link to the tool's doc

Reply:
We have podman guide in repo. I think adding orbstack guide is helpful to new contributer as it's not trival to run tests like other rust projects, we should lower the threshold for new contributers as much as possible.